### DISCUSSION TITLE
Era end rewards

### DISCUSSION THREAD

**Discussion by davidatwhiletrue** [casper-network/condor-info#48] [Q&A] [answered]
EraEndV2 shows two reward amounts per validator. What is each amount? Would it be possible to have more amounts in this array?

```
            "rewards": {
              "01032146b0b9de01e26aaec7b0d1769920de94681dbd432c3530bfe591752ded6c": [
                "41092396264634",
                "39200540522"
              ],
              "0126d4637eb0c0769274f03a696df1112383fa621c9f73f57af4c5c0fbadafa8cf": [
                "45495187485152",
                "79595918712"
              ],
              "0140afe8f752e5ff100e0189c080bc207e8805b3e5e82f792ec608de2f11f39f6c": [
                "42372557502281",
                "39392952373"
              ],
              "017536433a73f7562526f3e9fcb8d720428ae2d28788a9909f3c6f637a9d848a4b": [
                "47820101820165",
                "42465668964"
              ]
            },
```

**Comment by fizyk20** [comment, marked as answer]
Hi,

This has to do with how the new rewards scheme in Condor works. Now, every block can cite finality signatures for a few previous blocks and citing those signatures rewards the signer as well as the proposer of the citing block. The first few blocks in each era are allowed to cite signatures for the final blocks of the previous era. But since the signer of the block from the previous era might no longer be a validator in the current era, the rewards distribution code needs to know which parts of the full reward belong to which era. This is what this does: the first amount in the array is assigned to the current era, and the second is for the previous era.

At the moment, the arrays can have at most two values, because we put limits in place so that the rewards computation doesn't reach further than one era back.
